src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -628,126 +629,188 @@ protected AnalyzerActions GetOrCreateAnalyzerActions(DiagnosticAnalyzer analyzer
+            ImmutableArray<OperationAnalyzerAction> operationActions,
+            bool concurrent,
+            bool isEmpty)
+        {
+            _compilationStartActions = compilationStartActions;
+            _compilationEndActions = compilationEndActions;
+            _compilationActions = compilationActions;
+            _syntaxTreeActions = syntaxTreeActions;
+            _semanticModelActions = semanticModelActions;
+            _symbolActions = symbolActions;
+            _symbolStartActions = symbolStartActions;
+            _symbolEndActions = symbolEndActions;
+            _codeBlockStartActions = codeBlockStartActions;
+            _codeBlockEndActions = codeBlockEndActions;
+            _codeBlockActions = codeBlockActions;
+            _operationBlockStartActions = operationBlockStartActions;
+            _operationBlockEndActions = operationBlockEndActions;
+            _operationBlockActions = operationBlockActions;
+            _syntaxNodeActions = syntaxNodeActions;
+            _operationActions = operationActions;
+            _concurrent = concurrent;
+            IsEmpty = isEmpty;
+        }
+
+        public readonly int CompilationStartActionsCount { get { return _compilationStartActions.Length; } }
+        public readonly int CompilationEndActionsCount { get { return _compilationEndActions.Length; } }
+        public readonly int CompilationActionsCount { get { return _compilationActions.Length; } }
+        public readonly int SyntaxTreeActionsCount { get { return _syntaxTreeActions.Length; } }
+        public readonly int SemanticModelActionsCount { get { return _semanticModelActions.Length; } }
+        public readonly int SymbolActionsCount { get { return _symbolActions.Length; } }
+        public readonly int SymbolStartActionsCount { get { return _symbolStartActions.Length; } }
+        public readonly int SymbolEndActionsCount { get { return _symbolEndActions.Length; } }
+        public readonly int SyntaxNodeActionsCount { get { return _syntaxNodeActions.Length; } }
+        public readonly int OperationActionsCount { get { return _operationActions.Length; } }
+        public readonly int OperationBlockStartActionsCount { get { return _operationBlockStartActions.Length; } }
+        public readonly int OperationBlockEndActionsCount { get { return _operationBlockEndActions.Length; } }
+        public readonly int OperationBlockActionsCount { get { return _operationBlockActions.Length; } }
+        public readonly int CodeBlockStartActionsCount { get { return _codeBlockStartActions.Length; } }
+        public readonly int CodeBlockEndActionsCount { get { return _codeBlockEndActions.Length; } }
+        public readonly int CodeBlockActionsCount { get { return _codeBlockActions.Length; } }
+        public readonly bool Concurrent => _concurrent;
+        public bool IsEmpty { readonly get; private set; }
+        public readonly bool IsDefault => _compilationStartActions.IsDefault;
```

Comment:
This does not seem correct. Either it should be checking all fields, or we should have some sentinel `_isDefault` that is always set.

Reply:
There is no need to check all fields. There is no way to partially construct this type - it is either fully initialized or it is `default`.